js/recovery-sdk/src/index.ts

```diff
@@ -0,0 +1,80 @@
+import { keccak_256 } from '@noble/hashes/sha3';
+import { bytesToHex, randomBytes, concatBytes, hexToBytes, utf8ToBytes } from '@noble/hashes/utils';
+import { ContactType } from './types.js';
+import { standardizeContact } from './standardize.js';
+
+export * from './types.js';
+
+/**
+ * Generates a new Recovery Secret for a user
+ *
+ * @returns A new recovery secret string formatted for user consumption
+ */
+export function generateRecoverySecret(): string {
+  const unformatted = bytesToHex(randomBytes(32)).split('');
+  return [...Array(64 / 4)]
+    .map((_a) => unformatted.splice(0, 4).join(''))
+    .join('-')
+    .toUpperCase();
+}
+
+/**
+ * Gets the Intermediary Hashes `a` and `b` used to do the Recovery
+ *
+ * @param recoverySecret The string form of the Recovery Secret
+ * @param contactType Email = "0x00", Phone = "0x01"
+ * @param contact The raw contact string
+ * @returns { a: Hex String of H(s), b: H(s + sc) }
+ */
+export function getIntermediaryHashes(
+  recoverySecret: string,
+  contactType: ContactType,
+  contact: string
+): { a: string; b: string } {
```

Comment:
nit: might be nice here to type the return values as `` `0x${string}` ``. I find when dealing with hex strings, it's nice to be able to know at a glance whether the values you're getting back are `0x`-prefixed or not.

Reply:
Fixed